src/lib/Containers/Conversation.tsx

```diff
@@ -64,14 +81,20 @@ interface Props extends RelayProps {
 
 interface State {
   sendingMessage: boolean
+  isConnected: boolean
 }
 
 export class Conversation extends React.Component<Props, State> {
   constructor(props) {
     super(props)
+
+    // Assume if the component loads, connection exists (this way the banner won't flash unnecessarily)
     this.state = {
       sendingMessage: false,
+      isConnected: true,
     }
+
+    this.handleConnectivityChange = this.handleConnectivityChange.bind(this)
```

Comment:
One thing you can do to simplify component constructors and binding is to use ES7 class properties and fat arrows, which are supported by TypeScript and remove a lot of boiler plate.   

This:

```ts
constructor(props) {
  super(props)

  // Assume if the component loads, connection exists (this way the banner won't flash unnecessarily)
  this.state = {
    sendingMessage: false,
    isConnected: true,
  }

  this.handleConnectivityChange = this.handleConnectivityChange.bind(this)
}
```

can be simplified to 

```ts
export class Conversation extends React.Component<Props, State> {
  // Instance variables can be set directly on the class
  state = {
    sendingMessage: false,
    isConnected: true,
  }

  // And the fat arrow binds to the surrounding scope so no need for `.bind(this)`
  handleConnectivityChange = (event) => {
    ...
  }
}
```

I wish this were more widely known because it saves so many LOCs!